UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/EssWebhookService.cs

```diff
@@ -23,69 +31,157 @@ public class EssWebhookService : IEssWebhookService
-                        PartitionKey = cacheInfo.PartitionKey,
-                        RowKey = cacheInfo.RowKey,
-                        ETag = "*"
-                    };
-
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheDataFromTableStarted.ToEventId(), "Deletion started for Search and Download cache data from table:{cacheConfiguration.Value.FssSearchCacheTableName} for ProductName:{cellName} of BusinessUnit:{businessUnit} and BatchId:{cacheInfo.BatchId} and _X-Correlation-ID:{CorrelationId}", cacheConfiguration.Value.FssSearchCacheTableName, cellName, enterpriseEventCacheDataRequest.BusinessUnit, cacheTableData.BatchId, correlationId);
-                    await azureTableStorageClient.DeleteAsync(cacheTableData, cacheConfiguration.Value.FssSearchCacheTableName, storageConnectionString, essFulfilmentStorageconfig.Value.StorageContainerName);
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheDataFromTableCompleted.ToEventId(), "Deletion completed for Search and Download cache data from table:{cacheConfiguration.Value.FssSearchCacheTableName} for ProductName:{cellName} of BusinessUnit:{businessUnit} and BatchId:{cacheTableData.BatchId} and _X-Correlation-ID:{CorrelationId}", cacheConfiguration.Value.FssSearchCacheTableName, cellName, enterpriseEventCacheDataRequest.BusinessUnit, cacheTableData.BatchId, correlationId);
-
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheDataFromContainerStarted.ToEventId(), "Deletion started for Search and Download cache data from Blob Container for ProductName:{cellName} of BusinessUnit:{businessUnit} and BatchId:{cacheTableData.BatchId} and _X-Correlation-ID:{CorrelationId}", cellName, enterpriseEventCacheDataRequest.BusinessUnit, cacheTableData.BatchId, correlationId);
-                    await azureBlobStorageClient.DeleteCacheContainer(storageConnectionString, cacheTableData.BatchId);
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheDataFromContainerCompleted.ToEventId(), "Deletion completed for Search and Download cache data from Blob Container for ProductName:{cellName} of BusinessUnit:{businessUnit} and BatchId:{cacheTableData.BatchId} and _X-Correlation-ID:{CorrelationId}", cellName, enterpriseEventCacheDataRequest.BusinessUnit, cacheTableData.BatchId, correlationId);
+                    BatchId = enterpriseEventCacheDataRequest.BatchId,
+                    PartitionKey = cellName,
+                    RowKey = $"{editionNumber}|{updateNumber}|{enterpriseEventCacheDataRequest.BusinessUnit.ToUpper()}",
+                    Response = JsonConvert.SerializeObject(enterpriseEventCacheDataRequest)
+                };
+
+                await DeleteSearchAndDownloadCacheDataAsync(fssSearchResponse, storageConnectionString, correlationId);
+                if (enterpriseEventCacheDataRequest.Files != null && enterpriseEventCacheDataRequest.Files.Count > 0)
```

Comment:
Should we check if the action in line 91 has comepleted successfully here? Should DeleteSearchAndDownloadCacheDataAsync return a bool so we can ensure the old data has been deleted before we upload the new data

Reply:
we have implemented a conditional delete, where data will be deleted only if a matching record exists in cache table.
In case, if the deletion process fails it will throw exception and prevent subsequent insertion as well.